### PR TITLE
Adds ~70 more suicides

### PR DESCRIPTION
**Pull request by MrDoomBringer**
adds more depth to the game

:cl: MrDoomBringer
add: NanoTrasen's Creative Psychology Initiative has brought new training to all crewmembers to foster rapid, innovative problem-solving! You can now kill yourself in so many more ways!
/:cl:

why:
image

### CODE REVIEW

**Comment by Buggy123** [comment]
RIP the changelog.

**Comment by MrDoomBringer** [comment]
even the changelog commits suicide!

File: html/changelogs/AutoChangeLog-pr-34059.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: "ShiggyDiggyDo"
+delete-after: True
+changes: 
+  - tweak: "Syndicate experts were able to create more sophisticated copies of the nuclear authentication disk. While still unable to detonate the nuclear fission explosive with them, they're practically identical to the real disk if not examined closely."
```
Comment (praisenarsie) [inline html/changelogs/AutoChangeLog-pr-34059.yml]: illegal

Comment (MrDoomBringer) [reply html/changelogs/AutoChangeLog-pr-34059.yml]: how did this happen

Comment (MrDoomBringer) [reply html/changelogs/AutoChangeLog-pr-34059.yml]: will fix soon

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 2]
```diff
@@ -266,6 +266,11 @@
 	var/break_chance = 5 //Likely hood of smashing the chair.
 	var/obj/structure/chair/origin_type = /obj/structure/chair
 
+/obj/item/chair/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins hitting [user.p_them()]self with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(src,hitsound,50,1)
+	return BRUTELOSS
+
```
Comment (XDTM) [inline code/game/objects/structures/beds_chairs/chair.dm]: double

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 3]
```diff
@@ -348,6 +353,11 @@
 	materials = null
 	break_chance = 50
 
+/obj/item/chair/suicide_act(mob/living/carbon/user)
```
Comment (XDTM) [inline code/game/objects/structures/beds_chairs/chair.dm]: double

**Comment by nicbn** [comment]
Don't add a changelog entry for it please

File: code/game/objects/items/cosmetics.dm [hunk 4]
```diff
@@ -107,6 +107,10 @@
 	flags_1 = CONDUCT_1
 	w_class = WEIGHT_CLASS_TINY
 
+/obj/item/razor/suicide_act(mob/living/carbon/user)
```
Comment (MMMiracles) [inline code/game/objects/items/cosmetics.dm]: doesn't leave you bald, terrible suicide

**Comment by 81Denton** [comment]
>if (islizard(user))
> +		to_chat(user, "<span class='notice'>*HELL*</span>")

Beautiful

**Comment by Repukan** [comment]
is the handcuff suicide only when you're holding cuffs or while you're cuffed?

**Comment by Cyberboss** [comment]
For the love of god! Use commas, man!

File: code/game/objects/items/cosmetics.dm (hunk 4 above)
Comment (Cyberboss) [reply code/game/objects/items/cosmetics.dm]: Agreed

File: code/game/objects/items/cigs_lighters.dm [hunk 5]
```diff
@@ -490,6 +490,15 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	light_color = LIGHT_COLOR_FIRE
 	grind_results = list("iron" = 1, "welding_fuel" = 5, "oil" = 5)
 
+/obj/item/lighter/suicide_act(mob/living/carbon/user)
+	if (lit)
+		user.visible_message("<span class='suicide'>[user] begins holding \the [src]'s flame up to [user.p_their()] face! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, 'sound/items/welder.ogg', 50, 1)
+		return FIRELOSS
```
Comment (Cyberboss) [inline code/game/objects/items/cigs_lighters.dm]: Should self ignite or something

File: code/game/objects/items/devices/aicard.dm [hunk 6]
```diff
@@ -12,6 +12,10 @@
 	var/flush = FALSE
 	var/mob/living/silicon/ai/AI
 
+/obj/item/device/aicard/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] is trying to upload themselves into \the [src]! That's not going to work out well!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/aicard.dm]: Remove \the

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 7]
```diff
@@ -72,6 +72,14 @@ GLOBAL_LIST_EMPTY(PDAs)
 
 	var/underline_flag = TRUE //flag for underline
 
+/obj/item/device/pda/suicide_act(mob/living/carbon/user)
+	var/deathMessage = msg_input(user)
+	if (!deathMessage)
+		deathMessage = "I ded"
+	user.visible_message("<span class='suicide'>[user] is sending a message to The Grim Reaper! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/PDA/PDA.dm]: lowercase the

File: code/game/objects/items/devices/paicard.dm [hunk 8]
```diff
@@ -10,6 +10,10 @@
 	var/mob/living/silicon/pai/pai
 	resistance_flags = FIRE_PROOF | ACID_PROOF | INDESTRUCTIBLE
 
+/obj/item/device/paicard/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] is staring sadly at \the [src]! [user.p_they()] can't keep living without real human intimacy!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/paicard.dm]: again

File: code/game/objects/items/devices/radio/radio.dm [hunk 9]
```diff
@@ -42,6 +42,10 @@
 	var/const/FREQ_LISTENING = 1
 	//FREQ_BROADCASTING = 2
 
+/obj/item/device/radio/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] starts bouncing \the [src] off their head! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/radio/radio.dm]: again

File: code/game/objects/items/devices/scanners.dm [hunk 10]
```diff
@@ -18,6 +18,10 @@ GAS ANALYZER
 	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
 	materials = list(MAT_METAL=150)
 
+/obj/item/device/t_scanner/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to emit terahertz-rays into [user.p_their()] brain with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/scanners.dm]: again

File: code/game/objects/items/devices/scanners.dm [hunk 11]
```diff
@@ -74,6 +78,10 @@ GAS ANALYZER
 	var/scanmode = 0
 	var/advanced = FALSE
 
+/obj/item/device/healthanalyzer/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to analyze [user.p_them()]self with \the [src]! The display shows that [user.p_theyre()] dead!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/scanners.dm]: again

File: code/game/objects/items/devices/scanners.dm [hunk 12]
```diff
@@ -335,6 +343,10 @@ GAS ANALYZER
 	materials = list(MAT_METAL=30, MAT_GLASS=20)
 	grind_results = list("mercury" = 5, "iron" = 5, "silicon" = 5)
 
+/obj/item/device/analyzer/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to analyze [user.p_them()]self with \the [src]! The display shows that [user.p_theyre()] dead!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/devices/scanners.dm]: again

File: code/game/objects/items/extinguisher.dm [hunk 13]
```diff
@@ -45,6 +45,19 @@
 	create_reagents(max_water)
 	reagents.add_reagent("water", max_water)
 
+/obj/item/extinguisher/suicide_act(mob/living/carbon/user)
+	if (!safety && (src.reagents.total_volume >= 1))
+		user.visible_message("<span class='suicide'>[user] puts the nozzle to [user.p_their()] mouth. It looks like [user.p_theyre()] trying to extenguish the spark of life!</span>")
+		//playsound(src.loc, 'sound/effects/extinguish.ogg', 75, 1, -3)
```
Comment (Cyberboss) [inline code/game/objects/items/extinguisher.dm]: uncomment this and remove .loc

File: code/game/objects/items/extinguisher.dm [hunk 14]
```diff
@@ -45,6 +45,19 @@
 	create_reagents(max_water)
 	reagents.add_reagent("water", max_water)
 
+/obj/item/extinguisher/suicide_act(mob/living/carbon/user)
+	if (!safety && (src.reagents.total_volume >= 1))
+		user.visible_message("<span class='suicide'>[user] puts the nozzle to [user.p_their()] mouth. It looks like [user.p_theyre()] trying to extenguish the spark of life!</span>")
+		//playsound(src.loc, 'sound/effects/extinguish.ogg', 75, 1, -3)
+		src.afterattack(user,user)
```
Comment (Cyberboss) [inline code/game/objects/items/extinguisher.dm]: remove src

File: code/game/objects/items/extinguisher.dm [hunk 15]
```diff
@@ -45,6 +45,19 @@
 	create_reagents(max_water)
 	reagents.add_reagent("water", max_water)
 
+/obj/item/extinguisher/suicide_act(mob/living/carbon/user)
+	if (!safety && (src.reagents.total_volume >= 1))
+		user.visible_message("<span class='suicide'>[user] puts the nozzle to [user.p_their()] mouth. It looks like [user.p_theyre()] trying to extenguish the spark of life!</span>")
+		//playsound(src.loc, 'sound/effects/extinguish.ogg', 75, 1, -3)
+		src.afterattack(user,user)
+		return OXYLOSS
+	else if (safety && (src.reagents.total_volume >= 1))
```
Comment (Cyberboss) [inline code/game/objects/items/extinguisher.dm]: again

File: code/game/objects/items/extinguisher.dm [hunk 16]
```diff
@@ -45,6 +45,19 @@
...
+	if (!safety && (src.reagents.total_volume >= 1))
+		user.visible_message("<span class='suicide'>[user] puts the nozzle to [user.p_their()] mouth. It looks like [user.p_theyre()] trying to extenguish the spark of life!</span>")
+		//playsound(src.loc, 'sound/effects/extinguish.ogg', 75, 1, -3)
+		src.afterattack(user,user)
+		return OXYLOSS
+	else if (safety && (src.reagents.total_volume >= 1))
+		user.visible_message("<span class='warning'>[user] puts the nozzle to [user.p_their()] mouth... The safety's still on!</span>")
+		return SHAME
+	else
+		user.visible_message("<span class='warning'>[user] puts the nozzle to [user.p_their()] mouth... \The [src] is empty!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/extinguisher.dm]: again

File: code/game/objects/items/grenades/grenade.dm [hunk 17]
```diff
@@ -17,6 +17,14 @@
 	var/det_time = 50
 	var/display_timer = 1
 
+/obj/item/grenade/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] primes \the [src], then eats it! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/grenades/grenade.dm]: again

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Got a third done

File: code/game/objects/items/devices/multitool.dm [hunk 18]
```diff
@@ -159,4 +163,5 @@
 	desc = "An omni-technological interface."
 	icon = 'icons/obj/abductor.dmi'
 	icon_state = "multitool"
-	toolspeed = 0.1
\ No newline at end of file
+	toolspeed = 0.1
+
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/devices/multitool.dm]: remove this extra line

File: code/game/objects/items/devices/radio/headset.dm [hunk 19]
```diff
@@ -11,6 +11,10 @@
 	var/obj/item/device/encryptionkey/keyslot2 = null
 	dog_fashion = null
 
+/obj/item/device/radio/headset/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins putting \the [src]'s antenna up [user.p_their()] nose! It looks like [user.p_theyre()] trying to give [user.p_them()]self cancer!</span>")
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/devices/radio/headset.dm]: change nose to ```[prob(1) ? "butt" : "nose"]```

File: code/game/objects/items/handcuffs.dm [hunk 20]
```diff
@@ -110,7 +114,7 @@
 	breakouttime = 300 //Deciseconds = 30s
 	cuffsound = 'sound/weapons/cablecuff.ogg'
 	var/datum/robot_energy_storage/wirestorage = null
-
+
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/handcuffs.dm]: newline

File: code/game/objects/items/manuals.dm [hunk 21]
```diff
@@ -933,6 +933,10 @@
 	title = "Space Law"
 	page_link = "Space_Law"
 
+/obj/item/book/manual/wiki/security_space_law/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] pretends to read \the [src] intently... then promptly dies of laughter!</span>")
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/manuals.dm]: 🤣

File: code/game/objects/items/stacks/medical.dm [hunk 22]
```diff
@@ -127,6 +127,10 @@
 	self_delay = 20
 	max_amount = 12
 
+/obj/item/stack/medical/gauze/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] begins tightening \the [src] around [user.p_their()] neck! It looks like [user.p_they()] forgot how to use medical supplies!</span>")
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/medical.dm]: doko made futari de iru no ka na 
monomane sukiru ga mijuku na no 
tanoshii jikan ga ii no ka na 
honto no kimochi wa himitsu da yo

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 23]
```diff
@@ -123,6 +123,10 @@ GLOBAL_LIST_INIT(metal_recipes, list ( \
 	recipes = GLOB.metal_recipes
 	return ..()
 
+/obj/item/stack/sheet/metal/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins whacking themselves over the head with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: make this play an OOF sound

File: code/game/objects/items/cosmetics.dm (hunk 4 above)
Comment (MrDoomBringer) [reply code/game/objects/items/cosmetics.dm]: shit u right

File: code/game/objects/items/cigs_lighters.dm (hunk 5 above)
Comment (MrDoomBringer) [reply code/game/objects/items/cigs_lighters.dm]: i was thinking about that, but I was worried about the possibility of starting fires with something that cant start fires (even in a roundabout way)

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 23 above)
Comment (MrDoomBringer) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: like...an actual roblox oof sound?

File: code/game/objects/items/extinguisher.dm (hunk 13 above)
Comment (MrDoomBringer) [reply code/game/objects/items/extinguisher.dm]: whups, that shouldnt even be in there. sound is already handled on line 52

File: code/game/objects/structures/beds_chairs/chair.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/game/objects/structures/beds_chairs/chair.dm]: explain

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 23 above)
Comment (MoreRobustThanYou) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: OOF

File: code/game/objects/structures/beds_chairs/chair.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/game/objects/structures/beds_chairs/chair.dm]: explain further

**Comment by MrDoomBringer** [comment]
@Repukan, only when holding cuffs.

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 23 above)
Comment (MrDoomBringer) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: well...alright

File: code/game/objects/items/extinguisher.dm [hunk 24]
```diff
@@ -45,6 +45,18 @@
 	create_reagents(max_water)
 	reagents.add_reagent("water", max_water)
 
+/obj/item/extinguisher/suicide_act(mob/living/carbon/user)
+	if (!safety && (reagents.total_volume >= 1))
+		user.visible_message("<span class='suicide'>[user] puts the nozzle to [user.p_their()] mouth. It looks like [user.p_theyre()] trying to extenguish the spark of life!</span>")
```
Comment (cyclowns) [inline code/game/objects/items/extinguisher.dm]: extinguish

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 23 above)
Comment (Armhulen) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: @MrDoomBringer this is a bad idea walking roblox sounds are from an evil company i would remove it and never listen to that idiot again

File: code/game/objects/items/twohanded.dm [hunk 25]
```diff
@@ -286,6 +286,32 @@
 	var/brightness_on = 6 //TWICE AS BRIGHT AS A REGULAR ESWORD
 	var/list/possible_colors = list("red", "blue", "green", "purple")
 
+/obj/item/twohanded/dualsaber/suicide_act(mob/living/carbon/user)
+	if(wielded)
+		user.visible_message("<span class='suicide'>[user] begins spinning way too fast! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		
+		var/obj/item/bodypart/head/myhead = user.get_bodypart("head")//stole from chainsaw code
+		//INVOKE_ASYNC(src, .proc/jedi_spin, user)//sick spins
```
Comment (Armhulen) [inline code/game/objects/items/twohanded.dm]: ??

**Comment by PKPenguin321** [comment]
Dont line breaks mess up the change log?

File: code/game/objects/items/twohanded.dm (hunk 25 above)
Comment (MrDoomBringer) [reply code/game/objects/items/twohanded.dm]: whups meant to delete that

**Comment by LetterN** [comment]
but the newspaper suicide is still the best.

File: code/game/objects/items/cards_ids.dm [hunk 26]
```diff
@@ -19,6 +19,11 @@
 
 	var/list/files = list()
 
+/obj/item/card/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to swipe [user.p_their()] neck with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return BRUTELOSS
+
+
```
Comment (praisenarsie) [inline code/game/objects/items/cards_ids.dm]: remove a space

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 27]
```diff
@@ -72,6 +72,14 @@ GLOBAL_LIST_EMPTY(PDAs)
 
 	var/underline_flag = TRUE //flag for underline
 
+/obj/item/device/pda/suicide_act(mob/living/carbon/user)
+	var/deathMessage = msg_input(user)
+	if (!deathMessage)
+		deathMessage = "I ded"
```
Comment (praisenarsie) [inline code/game/objects/items/devices/PDA/PDA.dm]: ``i ded`` not ``I ded`` ;)

Comment (MrDoomBringer) [reply code/game/objects/items/devices/PDA/PDA.dm]: of course

**Comment by Qbopper** [comment]
just say that you added 60 new suicides don't spoil what they are or make the changelog a mess like that

File: code/game/objects/items/extinguisher.dm (hunk 13 above)
Comment (coiax) [reply code/game/objects/items/extinguisher.dm]: > extenguish

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 28]
```diff
@@ -102,6 +102,12 @@ GLOBAL_LIST_INIT(metal_recipes, list ( \
...
+	new /obj/structure/girder(loc)
+	user.gib()
```
Comment (coiax) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: I thought making easy to obtain gib suicides was bad for some reason?

**Comment by nicbn** [comment]
> just say that you added 60 new suicides don't spoil what they are or make the changelog a mess like that

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 28 above)
Comment (MrDoomBringer) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: outdated, i fucked up and pushed an older version of the branch

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 23 above)
Comment (MrDoomBringer) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: well...alright

**Comment by ShizCalev** [comment]
Isn't doing `\the [src]` redundant since you're not using the name var of the object?

File: code/modules/assembly/flash.dm [hunk 29]
```diff
@@ -15,6 +15,13 @@
 	var/cooldown = 0
 	var/last_trigger = 0 //Last time it was successfully triggered.
 
+/obj/item/device/assembly/flash/suicide_act(mob/living/user)
+	if (user.eye_blind)
+		user.visible_message("<span class='suicide'>[user] raises \the [src] up to [user.p_their()] eyes and activates it ... but [user.p_theyre()] blind!</span>")
+		return SHAME
+	user.visible_message("<span class='suicide'>[user] raises \the [src] up to [user.p_their()] eyes and activates it! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return FIRELOSS
```
Comment (coiax) [inline code/modules/assembly/flash.dm]: What if the flash is burnt out?

Comment (MrDoomBringer) [reply code/modules/assembly/flash.dm]: good call

**Comment by Qbopper** [comment]
👎 until the changelog is fixed

**Comment by MrDoomBringer** [comment]
@Qbopper by fixed do you mean put into a single line?

**Comment by optimumtact** [comment]
Yeah, any line breaks will fuck that up

**Comment by Qbopper** [comment]
@MrDoomBringer i would go even further and only say "60 new suicides"

leave the list for the PR where it can be read but not filling up the changelog in game

**Comment by MrDoomBringer** [comment]
alright guys i addressed your concerns

**Comment by Qbopper** [comment]
you left all of the individual entries in the changelog when i told you to not do that but i mean ok

**Comment by MrDoomBringer** [comment]
@optimumtact right

**Comment by optimumtact** [comment]
man qbopper stop minimodding

**Comment by lzimann** [comment]
Changelog really should be just "add new suicides/add x new suicides" and not a full list of everything, absolute awful to read a long list of items that you will forget in 5 minutes

**Review by imsxz — COMMENTED** [review]
nice

File: code/modules/assembly/timer.dm [hunk 30]
```diff
@@ -10,6 +10,16 @@
...
 
+/obj/item/device/assembly/timer/suicide_act(mob/living/user)
+	user.visible_message("<span class='suicide'>[user] looks at the timer and decides [user.p_their()] fate! It looks like [user.p_theyre()] going to commit suicide!</span>")
+	activate()//doesnt rely on timer_end to prevent weird metas where one person can control the timer and therefore someone's life. (maybe that should be how it works...)
+	addtimer(CALLBACK(src, .proc/manual_suicide, user), time*10)//kill yourself once the time runs out
+	return MANUAL_SUICIDE
+
+/obj/item/device/assembly/timer/proc/manual_suicide(mob/living/user)
+	user.visible_message("<span class='suicide'>[user]'s time is up!</span>")
+	user.adjustOxyLoss(200)
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/assembly/timer.dm]: nope

File: code/modules/assembly/signaler.dm [hunk 31]
```diff
@@ -13,6 +13,19 @@
...
+/obj/item/device/assembly/signaler/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] eats \the [src]! If it is signaled, [user.p_they()] will die!</span>")
+	playsound(src, 'sound/items/eatfood.ogg', 50, 1)
+	user.transferItemToLoc(src, user, TRUE)
+	suicider = user
+	return MANUAL_SUICIDE
+
+/obj/item/device/assembly/signaler/proc/manual_suicide(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user]'s \the [src] recieves a signal, killing them instantly!</span>")
+	user.adjustOxyLoss(200)//it sends an electrical pulse to their heart, killing them. or something.
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/assembly/signaler.dm]: nope

File: code/modules/assembly/flash.dm [hunk 32]
```diff
@@ -15,6 +15,16 @@
...
 	var/last_trigger = 0 //Last time it was successfully triggered.
 
+/obj/item/device/assembly/flash/suicide_act(mob/living/user)
+	if (crit_fail)
+		user.visible_message("<span class='suicide'>[user] raises \the [src] up to [user.p_their()] eyes and activates it ... but its burnt out!</span>")
+		return SHAME
+	else if (user.eye_blind)
+		user.visible_message("<span class='suicide'>[user] raises \the [src] up to [user.p_their()] eyes and activates it ... but [user.p_theyre()] blind!</span>")
+		return SHAME
+	user.visible_message("<span class='suicide'>[user] raises \the [src] up to [user.p_their()] eyes and activates it! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/assembly/flash.dm]: eeeeeeh ok

File: code/game/objects/items/storage/boxes.dm [hunk 33]
```diff
@@ -36,6 +36,15 @@
 	. = ..()
 	update_icon()
 
+/obj/item/storage/box/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] puts [user.p_their()] head into \the [src], and begins closing it! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/game/objects/items/storage/boxes.dm]: right

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 23 above)
Comment (MoreRobustThanYou) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: wtf @Armhulen

**Pull request merged**